### ISSUE TITLE
MW2.0: platform update

### ISSUE THREAD

**Issue by cstockloew** [universAAL/middleware#290]
_Originally Opened: @michelegirolami (2013-04-16 09:28:47_)
_Originally Closed: 2013-08-02 17:07:32_

Check if it is possible to install a Karaf feature that contains artifact already present but with a minor version.

--

From: _this issue has been automatically imported from our old issue tracker_

**Comment by cstockloew** [comment]
_Original author: Giancarlo Riolo (2013-08-02 17:07:32)_

Given a karaf feature repository we verified that it is possible to upgrade to a newest version: 
1) bundles can have different version w.r.t. the previous ones
2) the new Karaf feature can define less bundles w.r.t the previous version

We found that we can achieve such behaviour by adopting the KAR archives (JAR/ZIP). A Kar is a file that contains a feature inside. It is possible to declare different versions for every bundle, and even other bundles.

Constraints:

1)we have to use same .KAR name. Then inside the .KAR file we can put a different version of the feature. For example:
MW.kar contains feature1.0. As soon as we have a new feature ex  feature2.0 we must wrap it into MW.kar.

2)we can create the .KAR file in 2 ways: handmade, or using maven plugin (prefered)

here there are described both methods
http://karaf.apache.org/manual/latest/users-guide/kar.html
